thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/views/DimensionTreeMapView.js

```diff
@@ -182,21 +265,27 @@ DimensionTreeMapView.prototype = {
       });
 
     }
+    // anchor page to dimension tree map if exists
+    $("#dimension-tree-map-placeholder").get(0).scrollIntoView()
+  },
+
+  calculateBaselineDate(dateType, offset) {
+    const baseDate = this.dimensionTreeMapModel[dateType] || moment();
+    return baseDate.clone().subtract(offset, 'days');
   },
 
   setupListenersForMode : function () {
-    var self = this;
-    $("#percent_change a").click(self, function() {
-      self.dimensionTreeMapModel.heatmapMode = "percentChange";
-      self.render();
+    $("#percent_change a").click(() => {
+      this.dimensionTreeMapModel.heatmapMode = "percentChange";
+      this.renderTreemapSection();
     });
-    $("#change_in_contribution").click(self, function() {
-      self.dimensionTreeMapModel.heatmapMode = "contributionChange";
-      self.render();
+    $("#change_in_contribution").click(() => {
+      this.dimensionTreeMapModel.heatmapMode = "contributionChange";
+      this.renderTreemapSection();
     });
-    $("#contribution_to_overall_change").click(self, function() {
-      self.dimensionTreeMapModel.heatmapMode = "contributionToOverallChange";
-      self.render();
+    $("#contribution_to_overall_change").click(() => {
+      this.dimensionTreeMapModel.heatmapMode = "contributionToOverallChange";
+      this.renderTreemapSection();
```

Comment:
Yes, will definitely make this DRYer at the next iteration (with ember). 
I touched those line mainly to remove the `self` horror